test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -112,4 +120,44 @@ UCS_TEST_P(test_ucp_sockaddr, listen) {
     tag_send(receiver(), sender());
 }
 
+UCS_TEST_P(test_ucp_sockaddr, err_handle) {
+
+    struct sockaddr_in listen_addr;
+    err_handler_count = 0;
+
+    get_listen_addr(&listen_addr);
+
+    ucs_status_t status = receiver().listen((const struct sockaddr*)&listen_addr,
+                                            sizeof(listen_addr));
+    if (status == UCS_ERR_INVALID_ADDR) {
+        UCS_TEST_SKIP_R("cannot listen to " + ucs::sockaddr_to_str(&listen_addr));
+    }
+
+    /* make the client try to connect to a non-existing port on the server side */
+    listen_addr.sin_port = 1;
+
+    ucp_ep_params_t ep_params = ucp_test::get_ep_params();
+    ep_params.field_mask      |= UCP_EP_PARAM_FIELD_FLAGS |
+                                 UCP_EP_PARAM_FIELD_SOCK_ADDR |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLER_CB |
+                                 UCP_EP_PARAM_FIELD_USER_DATA;
+    ep_params.err_mode         = UCP_ERR_HANDLING_MODE_PEER;
+    ep_params.err_handler_cb   = err_handler_cb;
+    ep_params.user_data        = reinterpret_cast<void*>(this);
+    ep_params.flags            = UCP_EP_PARAMS_FLAGS_CLIENT_SERVER;
+    ep_params.sockaddr.addr    = (struct sockaddr*)&listen_addr;
+    ep_params.sockaddr.addrlen = sizeof(listen_addr);
+    wrap_errors();
+    sender().connect(&receiver(), ep_params);
+    /* allow for the unreachable event to arrive before restoring errors */
+    sleep (3);
```

Comment:
better to put wait_for_flag() here instead of sleep, because wait_for_flag() will wait for at most 10sec anyway, so no need to increase the test time by 3 seconds
also might need to increase the wait time if running on valgrind